docs/en/stores/supplemental.md

```diff
@@ -0,0 +1,425 @@
+# State
+
+In modern browsers a `state` object is passed as part of the CommandRequest. Any modification to this object is translated to the appropriate patch operations and applied against the store.
+
+```ts
+import { createCommandFactory } from '@dojo/framework/stores/process';
+import { State } from './interfaces';
+import { remove, replace } from '@dojo/framework/stores/state/operations';
+
+const createCommand = createCommandFactory<State>();
+
+const addUser = createCommand<User>(({ payload, state }) => {
+	const currentUsers = state.users.list || [];
+	state.users.list = [...currentUsers, payload];
+});
+```
+
+Note that attempting to access state is not supported in IE and will immediately throw an error.
+
+# StoreProvider
+
+## StoreProvider API
+
+The StoreProvider accepts three properties
+
+-   `renderer`: A render function that has the store injected in order to access state and pass processes to child widgets.
+-   `stateKey`: The key of the state in the registry.
+-   `paths` (optional): A function to connect the provider to sections of the state.
+
+## Invalidation
+
+The `StoreProvider` has two main ways to trigger invalidation and cause a rerender.
+
+1.  The recommended approach is to register `path`s by passing the `paths` property to the provider to ensure invalidation will only occur when state you are interested in changes.
+1.  A catch-all when no `path`s are defined for the container, it will invalidate when _any_ data changes in the store.
+
+## Pre-typing
+
+A pre-typed container can be created by extending the standard `StoreProvider` and passing the `State` type as a generic.
```

Comment:
I don't think we should really recommend doing this? Soon these will be converted to functional widgets and it won't be possible - at that point we'd be able to provide a util function that can be used to create a "typed" provider.